add ratio_unfinished param to allow jobs failed or discarded in a submission

shazj99 opened the pull request:
In order to speed up dpgen jobs, it may be practicable to accelerate FP stage by just skipping slow jobs or waiting finished asynchronously.

While running dpgen jobs, we found the duration time of FP phrase is a large part in each iteration. This is because the DFT computing for some candidates are really hard and time consuming, and we need to wait all those long tail to be finished before going to the next iteration. We found that the proportion of those candidates is very small, which may less than 1%.

So we may add new params "ratio_unfinished" to optimize the execution: if most of fp jobs have finished, we can directly discard the remaining jobs and go to the next iteration. We think it is acceptable since the ratio is very small. Further more, we can also enable async check to wait these jobs finished and download results asynchronously, but this step is need to be done in dpgen. Now we modify dpdispatcher as the first step.

Following is our test number, which can significantly saving time:
![image](https://user-images.githubusercontent.com/4508078/138858125-cc4f1935-b518-4ab5-a006-409a64631888.png)

njzjz commented on dpdispatcher/submission.py:
```diff
@@ -169,11 +169,16 @@ def run_submission(self, *, exit_on_submit=False, clean=True):
             self.check_all_finished()
             self.handle_unexpected_submission_state()
 
+        ratio_failure = self.resources.kwargs.get('ratio_failure', 0)
```
Why is it called `ratio_failure`? Isn't it `ratio_unfinished`?

shazj99 replied:
`ratio_unfinished` is okay for me. I use `ratio_failure` since I saw there is a same param in dpgen, which accomplishes the same thing:
https://github.com/deepmodeling/dpgen/blob/master/dpgen/dispatcher/DispatcherList.py#L48

njzjz reviewed (changes requested):
Please add argument to `Resources.arginfo`.
https://github.com/deepmodeling/dpdispatcher/blob/d792862c00072ad96d1bbe3c0dc029ccc20ff22a/dpdispatcher/submission.py#L763

njzjz commented:
Also, it may cause some problems to delete files directly without killing tasks.

njzjz commented:
Could you please add argument to [`Resources.arginfo`](https://github.com/deepmodeling/dpdispatcher/blob/d792862c00072ad96d1bbe3c0dc029ccc20ff22a/dpdispatcher/submission.py#L763)?

shazj99 commented:
@njzjz I've updated the patch according to your comments. Please review it again.

felix5572 commented:
> @felix5572 any ideas why the unit test fails?

There is a new field ratio_unfinished in the json file. This caused the unittest failed

shazj99 commented:
> > @felix5572 any ideas why the unit test fails?
> 
> There is a new field `ratio_unfinished` in the json file. This caused the unittest failed

Thanks for the review.
I've fixed and updated it. Pls take a look at again.  @njzjz @felix5572
